e2e/machine_migration_test.go

```diff
@@ -347,3 +618,76 @@ func cleanupMachineResources(ctx context.Context, cl client.Client, capiMachines
 		mapiframework.WaitForMachinesDeleted(cl, m)
 	}
 }
+
+func verifyCAPIMachineExists(cl client.Client, machineName string) {
+	Eventually(func() error {
+		_, err := capiframework.GetMachine(cl, machineName, capiframework.CAPINamespace)
+		return err
+	}, capiframework.WaitLong, capiframework.RetryLong).Should(Succeed(), "CAPI Machine should exist")
+}
+
+func verifyFieldModificationPrevented(cl client.Client, machineName string, fieldName string, modifyFunc func(*machinev1beta1.Machine)) {
+	By(fmt.Sprintf("Attempting to modify %s in MAPI Machine", fieldName))
+
+	currentMachine, err := mapiframework.GetMachine(cl, machineName)
+	if err != nil {
+		Fail(fmt.Sprintf("Failed to get current machine: %v", err))
+	}
+	updatedMachine := currentMachine.DeepCopy()
+	modifyFunc(updatedMachine)
+
+	err = cl.Update(ctx, updatedMachine)
+	Expect(err).To(HaveOccurred())
+	Expect(err.Error()).To(ContainSubstring("ValidatingAdmissionPolicy 'machine-api-machine-vap' with binding 'machine-api-machine-vap' denied request"))
+}
+
+func verifyAnnotationModificationPrevented(cl client.Client, machineName string) {
```

Comment:
We could use the above for this? `verifyFieldModificationPrevented`? With the changes from the above comment i don't think we need a helperfunc specific to annotation changes :)